ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OAutoML.scala

```diff
@@ -73,14 +74,14 @@ class H2OAutoML(override val uid: String) extends Estimator[H2OMOJOModel]
     spec.input_spec.weights_column = getWeightCol()
     spec.input_spec.ignored_columns = getIgnoredCols()
     spec.input_spec.sort_metric = getSortMetric()
-    spec.build_models.exclude_algos = if (getExcludeAlgos() == null) null else Array(getExcludeAlgos(): _*)
-    spec.build_models.include_algos = if (getIncludeAlgos() == null) null else Array(getIncludeAlgos(): _*)
+    spec.build_models.exclude_algos = if (getExcludeAlgos() == null) null else Array(getExcludeAlgos().map(Algo.valueOf): _*)
```

Comment:
Do we need to support `null` values for `excludeAlgos` and `includeAlgos`. Wouldn't empty arrays play the same role?

Reply:
Unfortunately yes, empty array is valid value for AutoML. If include_algos is set to empty array then no algos are selected.

Reply:
Ok, thanks!

Reply:
I was playing with a thought of setting includeAlgos by default to all & excludeAlgos to empty array, but AutoML complains as these 2 options are mutually exclusive and we can't programmatically decide which to choose.


One solution to avoid null on the API which would be possible is to do the following:

By default, have:
```
includeAlgos = [ALL]
excludeAlgos = null
```

After calling setExcludeAlgos(), set includeAlgos to null, print out warning and set excludeAlgos.

The same for the opposite way. What do you think?

Reply:
I think we can leave it as it is :-)

Reply:
Cool, I will at least create a JIRA for it, so we can think of it later in scope of a smaller change